controller/test/frame_detector/frame_detector_test.cpp

```diff
@@ -0,0 +1,144 @@
+#include "frame_detector.h"
+#include "proto_traits.h"
+#include "soft_rx_buffer.h"
+#include "gtest/gtest.h"
+
+constexpr uint32_t LEN = 20;
+
+constexpr FrameDetector<SoftRxBuffer<LEN>, LEN>::State STATE_LOST =
```

Comment:
You can simply say `using State = FrameDetector<SoftRxBuffer<LEN>, LEN>::State;`